Sources/SWBUtil/FSProxy.swift

```diff
@@ -28,82 +29,86 @@ public import struct Foundation.URL
 public import struct Foundation.URLResourceKey
 public import struct Foundation.URLResourceValues
 public import struct Foundation.UUID
-
-#if os(Windows)
-// Windows' POSIX layer does not have S_IFLNK, so define it.
-// We only need it for PseudoFS.
-fileprivate let S_IFLNK: Int32 = 0o0120000
+public import struct Foundation.FileAttributeType
+public import struct Foundation.FileAttributeKey
+public import struct Foundation.TimeInterval
+public import class Foundation.NSDictionary
+#if os(macOS)
+import struct ObjectiveC.ObjCBool
 #endif
 
 /// File system information for a particular file.
 ///
 /// This is a simple wrapper for stat() information.
 public struct FileInfo: Equatable, Sendable {
-    public let statBuf: stat
+    public let fileAttrs: [FileAttributeKey: any Sendable]
 
-    public init(_ statBuf: stat) {
-        self.statBuf = statBuf
+    public init(_ fileAttrs: [FileAttributeKey: any Sendable]) {
+        self.fileAttrs = fileAttrs
+    }
+
+    func _readFileAttributePrimitive<T: BinaryInteger>(_ value: Any?, as type: T.Type) -> T? {
+        guard let value else { return nil }
+        if let exact = value as? T {
+            return exact
+        } else if let binInt = value as? (any BinaryInteger), let result = T(exactly: binInt) {
+            return result
+        }
+        return nil
     }
 
     public var isFile: Bool {
-        #if os(Windows)
-        return (statBuf.st_mode & UInt16(ucrt.S_IFREG)) != 0
-        #else
-        return (statBuf.st_mode & S_IFREG) != 0
-        #endif
+        return (fileAttrs[.type] as! FileAttributeType == .typeRegular)
```

Comment:
Can you express this as `return fileAttrs[.type] == FileAttributeType.typeRegular` to avoid the force cast?